test/integration/edition_edit_test.rb

```diff
@@ -108,53 +94,149 @@ class EditionEditTest < IntegrationTest
         assert page.has_text?(/English/)
       end
     end
+  end
 
-    context "unpublish tab" do
+  context "unpublish tab" do
+    context "do not have required permissions" do
       setup do
-        click_link("Unpublish")
+        login_as(FactoryBot.create(:user, name: "Stub User"))
+        edition = FactoryBot.create(:guide_edition, state: "draft")
+        visit edition_path(edition)
       end
 
-      should "show 'Unpublish' header and 'Continue' button" do
-        within :css, ".gem-c-heading" do
-          assert page.has_text?("Unpublish")
-        end
-        assert page.has_button?("Continue")
+      should "not show unpublish tab when user is not govuk editor" do
+        assert page.has_no_text?("Unpublish")
       end
+    end
 
-      should "show 'cannot be undone' banner" do
-        assert page.has_text?("If you unpublish a page from GOV.UK it cannot be undone.")
+    context "has required permissions" do
+      setup do
+        login_as(FactoryBot.create(:user, :govuk_editor, name: "Stub User"))
+        edition = FactoryBot.create(:guide_edition, state: "draft")
+        visit edition_path(edition)
       end
 
-      should "show 'Redirect to URL' text, input box and example text" do
-        assert page.has_text?("Redirect to URL")
-        assert page.has_text?("For example: https://www.gov.uk/redirect-to-replacement-page")
-        assert page.has_css?(".govuk-input", count: 1)
+      context "when state is publish" do
+        setup do
+          @edition = FactoryBot.create(:guide_edition, state: "published")
```

Comment:
We should be consistent about using `@edition` vs `edition`.

Reply:
Again, this is @edition because there is an assert on line:145 which will not work with edition. This test file would grow and I am not sure moving everything to @edition adds any benefit other than consistency, which I dont think is a huge win in this file at the moment.

Reply:
Ok, took a step back to think about how to improve this, think now I have a better way to deal with this.

Reply:
have now, updated the file and made a few improvements which should also improve this issue

Reply:
Again, this was @edition because there was an assert on line:145 which will not work with edition. 
Also, this file was not ready for 2nd review yet as I was still working on it after our discussion after 1st review. I now have refactored the test file and think refactor should fix this.